lib/model.js

```diff
@@ -4667,6 +4666,13 @@ function populate(model, docs, options, callback) {
       populateOptions.strictPopulate = model.base.options.strictPopulate;
     }
   }
+  let originalLocalField = '';
+  if (model.schema.virtuals[populateOptions.path]) {
+    originalLocalField = model.schema.virtuals[populateOptions.path].options.localField;
+  }
+  if (populateOptions.localField) {
+    model.schema.virtuals[populateOptions.path].options.localField = populateOptions.localField;
```

Comment:
Modifying the virtual options is a bit of a red flag. This will likely mean that future `populate()` calls will use the new `localField` by default. We'll need to avoid doing this somehow.